src/fiatconnect/ReviewScreen.test.tsx

```diff
@@ -102,9 +108,57 @@ describe('ReviewScreen', () => {
       ])
     })
 
+    it('shows expired dialog when quote is expired', async () => {
+      const expireMs = -100
+      const mockProps = getProps(CICOFlow.CashOut, false, CryptoType.cUSD, expireMs)
+      const quote = _.cloneDeep(mockFiatConnectQuotes[1]) as FiatConnectQuoteSuccess
+      quote.quote.guaranteedUntil = new Date(Date.now() + expireMs).toISOString()
+      const store = createMockStore({
+        fiatConnect: {
+          quotes: [quote],
+        },
+      })
+      const { getByTestId, queryByTestId } = render(
+        <Provider store={store}>
+          <FiatConnectReviewScreen {...mockProps} />
+        </Provider>
+      )
+
+      expect(queryByTestId('expiredQuoteDialog')?.props.visible).toEqual(true)
+
+      await fireEvent.press(getByTestId('expiredQuoteDialog/PrimaryAction'))
+
+      expect(store.getActions()).toEqual([
+        fetchFiatConnectQuotes({
+          flow: CICOFlow.CashOut,
+          digitalAsset: CiCoCurrency.CUSD,
+          cryptoAmount: 100,
+          providerIds: [mockFiatConnectQuotes[1].provider.id],
+        }),
+      ])
+      expect(queryByTestId('expiredQuoteDialog')?.props.visible).toEqual(true)
```

Comment:
Shouldn't this be not visible after the user has clicked the try again button?

I think to see the UI change based on internal state changes you'll need to use [rerender](https://callstack.github.io/react-native-testing-library/docs/api/#rerender)

Reply:
Ideally yes, but unfortunately I was having a fair bit of trouble mocking fetching an updated quote. I think what happens is that the showingExpiredQuoteDialog state is set to false after the fireEvent, but the only quote present is the expired quote. This means that showingExpiredQuoteDialog is set to true again. I might just delete that expect statement